pkg/cmd/cli/backuplocation/set.go

```diff
@@ -104,36 +105,40 @@ func (o *SetOptions) Run(c *cobra.Command, f client.Factory) error {
 		}
 	}
 
-	location := &velerov1api.BackupStorageLocation{}
-	err = kbClient.Get(context.Background(), kbclient.ObjectKey{
-		Namespace: f.Namespace(),
-		Name:      o.Name,
-	}, location)
+	defalutBSLs, err := storage.GetDefaultBackupStorageLocations(context.Background(), kbClient, f.Namespace())
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	if o.DefaultBackupStorageLocation {
-		// There is one and only one default backup storage location.
-		// Disable the origin default backup storage location.
-		locations := new(velerov1api.BackupStorageLocationList)
-		if err := kbClient.List(context.Background(), locations, &kbclient.ListOptions{Namespace: f.Namespace()}); err != nil {
-			return errors.WithStack(err)
+	if !o.DefaultBackupStorageLocation { // unset default backup storage location
+		if len(defalutBSLs.Items) == 0 {
```

Comment:
Seems the `set` command only targets one specific `BSL`, why do you need to check others?

Reply:
this checking logic is for when the user uses the command `velero backup-location set $BSL --default=false`
to unset the default for one specific `BSL`

Reply:
I've removed the unset logic check, we allow the user unset one specific BSL repeatedly